### ISSUE TITLE
Adam weight decay optimizer for TF 2.0

### ISSUE THREAD

**Issue by artitw** [tensorflow/tensorflow#34341]
**System information**
- TensorFlow version (you are using): 2.0
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
I am looking to migrate some optimization code I forked from BERT to TF 2.0. The optimization involves the Adam weight decay optimizer and is implemented in TF 1.15 according to
https://github.com/artitw/text2class/blob/master/text2class/optimization.py

**Will this change the current api? How?**
For the TF 2.0 API, we can specify the Adam optimizer like below. However, it is unclear how the weight decay component can be implemented as it requires keeping track of the global step. 
```
model.compile(optimizer=tf.keras.optimizers.Adam(
    learning_rate=2e-5,
    beta_1=0.9,
    beta_2=0.999,
    epsilon=1e-6,),
                loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
                metrics=[tf.keras.metrics.SparseCategoricalAccuracy(name="acc")])
```

**Who will benefit with this feature?**
Researchers and engineers working with BERT and BERT-based models will benefit. The original BERT repo has almost 20k stars, but it has yet to be migrated to TF 2.0.

**Any Other info.**
If the capability to achieve what I describe above already exists, please let me know!

**Comment by PhilJd** [comment]
Hi @artitw,
sorry, I just saw your post now. AdamW has been migrated to [tf-addons](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/weight_decay_optimizers.py).
Regarding the global step: I'm not too familiar with the keras interface but the weight decay extension is just a wrapper around the vanilla Adam. The weight decay part doesn't need the global step. If you're using a learning rate schedule that depends on the global step, you should apply the same schedule to the weight decay.

Cheers,
Phil

**Comment by artitw** [comment]
Hi Phil,

Thanks for the update and tips! I'll close this now.

Best,
Art
